src/core/processing/qgsprocessingutils.cpp

```diff
@@ -846,9 +846,11 @@ QgsFeatureSink *QgsProcessingUtils::createFeatureSink( QString &destination, Qgs
 
         // use destination string as layer name (eg "postgis:..." )
         if ( !layerName.isEmpty() )
+        {
           uri += QStringLiteral( "|layername=%1" ).arg( layerName );
-        // update destination to generated URI
-        destination = uri;
+          // update destination to generated URI
+          destination = uri;
```

Comment:
Won't this still be broken when layername is non-empty?

Reply:
Actaully is my interpretation correct and layer name is only set for ogr providers?

Reply:
> Actaully is my interpretation correct and layer name is only set for ogr providers?

That's also what I think. layerName is meaningless in case of a Postgis provider, and so it's better not to change the destination. 

I hesitated with always restoring the destination according to providerKey and uri this way

```c++
destination = providerKey + "://" + uri
```

But, I'm afraid it could possibly break something else in other part of the code. So I choose the less intrusive way of fixing.

Reply:
Ok, I'm fine with that. Can you confirm that you've tested saving to a geopackage db and that there's no issues there?

Reply:
just tested on a gpkg file output and working